dotcom-rendering/src/lib/grid.ts

```diff
@@ -0,0 +1,164 @@
+// ----- Imports ----- //
+
+import { from } from '@guardian/source-foundations';
+
+// ----- Columns & Lines ----- //
+
+/**
+ * Named CSS grid lines, based on the three columns commonly used for Guardian
+ * layouts.
+ */
+type Line =
```

Comment:
This API is already used in AR for Immersives and Galleries: https://github.com/guardian/dotcom-rendering/pull/4874

Reply:
We went with a CSS only version for the time being.